src/main/java/someboty/Managers/taskManager.java

```diff
@@ -0,0 +1,145 @@
+            return "Wow! You have no recorded task! Peek laziness here.";
+        }
+
+        String response = "Here are the tasks:\n";
+
+        int index = 0;
+        for (Task task : taskList) {
+            index++;
+            response += String.format("%d. %s\n",
+                            index,
+                            task
+                            );
+        }
+
+        return response;
+    }
+
+    /**
+     * Set the completion status of a task in the task list.
+     * Status is set to true if the task is marked as completed, else set to false.
+     * @param index The i-th position of the task in the list. (1-indexed)
+     * @param status The completion status of the task to set.
+     * @return The task in the i-th position on the list.
+     * @throws InputException If index argument is out of task list's range.
+     */
+    protected Task setTaskStatus(int index, boolean status) throws InputException {
+        try {
+            this.taskList.get(index).setStatus(status);
+            return this.taskList.get(index);
+
+         } catch (IndexOutOfBoundsException e) {
+            throw new InputException(">>> Bruh, there ain't no task " + String.valueOf(index + 1));
+        }
+    }
+
+    /**
+     * Deletes a task from the task list.
+     * @param index The position of the task to be deleted (1-index).
+     * @return The task that has just been deleted from the list.
+     * @throws InputException If index argument is out of task list's range.
+     */
+    protected Task deleteTask(int index) throws InputException{
+        try {
+            Task removedTask = taskList.remove(index);
+            return removedTask;
+
+        } catch (IndexOutOfBoundsException e) {
+            throw new InputException(">>> Bruh, there ain't no task " + String.valueOf(index + 1));
+        }
+    }
+
+    /**
+     * Removes all existing tasks in the list.
+     */
+    protected void clear() {
+        this.taskList.clear();
+    }
+
+    /**
+     * Adds a new task into the task list.
+     * The type argument is represented using the first character in the full task type.
+     * @param type The type of the new task.
+     * @param description Details of the new task, including its name and (if any) dates.
+     * @return Successfully created new task.
+     */
+    protected Task addTask(char type, String description) {
+        Task newTask = Task.createTask(type, description);
+        this.taskList.add(newTask);
+        return newTask;
+    }
+
+    /**
+     * Finds all existing tasks that has matching names with a given substring.
+     * The substring will be case insensitive.
+     * @param substring Substring to match with all task names.
+     * @return A list of tasks with names that contain the substring.
+     */
+    protected ArrayList<Task> findTasks(String substring) {
+        ArrayList<Task> matchList = new ArrayList<>();
```

Comment:
Would this benefit from a clearer variable name? Without reading the Javadocs, I was unsure whether the list contains Task variables that needs to be matched, or whether it contains Task variables that matches a given string.